### PR TITLE
DRAFT: MCR-4138: update SubmissionType  and ContractDetails pages to use contract API

### PR DESCRIPTION
**Pull request by pearl-truss**
## Summary

This PR updates the SubmissionType page to use fetchContract, createContract and updateContractDraftRevision apis

## To Dos

- [x] fix cypress
- [x] fix skipped unit test

#### Related issues

https://jiraent.cms.gov/browse/MCR-4138

#### Test cases covered

Existing test updated to use mock contract API

## QA guidance
- Double check the UI for both creating a new submission as well as editing an existing draft submission (there were some issues I think I fully resolved but want other eyes on the draft portion especially with the page values for a draft displaying as blank instead of being populated with values returned from the fetchContract api)

### CODE REVIEW

File: services/app-web/src/pages/StateSubmission/New/NewStateSubmissionForm.test.tsx [hunk 1]
```diff
@@ -24,13 +24,13 @@ describe('NewStateSubmissionForm', () => {
             screen.getByRole('form', { name: 'New Submission Form' })
         ).toBeInTheDocument()
     })
-    it('displays generic error banner when creating new health plan package fails', async () => {
+    it.skip('displays generic error banner when creating new health plan package fails', async () => {
```
Comment (haworku) [inline services/app-web/src/pages/StateSubmission/New/NewStateSubmissionForm.test.tsx:27]: did you mean to skip here?

File: services/app-web/src/hooks/useContractForm.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    })
+    const [createFormData] = useCreateContractMutation()
+
+    const createDraft: UseContractForm['createDraft']  = async (
+        input: CreateContractInput
+    ): Promise<Contract | Error> => {
+        setShowPageErrorMessage(false)
+        const {populationCovered,programIDs,riskBasedContract, submissionType, submissionDescription, contractType} = input
+        if(populationCovered === undefined || contractType === undefined) {
+            return new Error('wrong')
```
Comment (haworku) [inline services/app-web/src/hooks/useContractForm.ts:65]: I think we could fix this now - I believe this should be handled with form validations

File: services/app-web/src/hooks/useContractForm.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+} from '../gen/gqlClient'
+import { recordJSException } from '../otelHelpers'
+import { handleApolloError } from '../gqlHelpers/apolloErrors'
+import { ApolloError } from '@apollo/client'
+import type { InterimState } from '../pages/StateSubmission/ErrorOrLoadingPage'
+
+
+type  UseContractForm = {
+    draftSubmission?: Contract
+    unlockInfo?: UpdateInformation
```
Comment (haworku) [inline services/app-web/src/hooks/useContractForm.ts:20]: Is this needed when we also returning to draft Submission? I'm a bit confused why this is surfaced as well

**Comment by JasonLin0991** [comment]
I manually ran through the review app and here are some things I found

- Save as draft does not save all the fields on the contract details page. Uploading a file will save the fields.
- System error on Submission type page
   - Completely fill out the contract details page.
   - Save as draft.
   - Click back into the submission
   - On the submission type page, clicking `Continue` or `Save as draft` will cause a system error with a 400 response from the API
- Unlocking a submission, and then viewing it as a state user will cause a System error when clicking `Continue` or `Save as draft`.
- When viewing a draft or unlocked submission, the submission type page will show unfilled fields for a second before filling them in with the data from the API.

These issues are probably whats causing the Cypress failures. Especially this one `When viewing a draft or unlocked submission, the submission type page will show unfilled fields for a second before filling them in with the data from the API.`

**Review by JasonLin0991 — COMMENTED** [review]
I manually ran through the review app and here are some things I found

Save as draft does not save all the fields on the contract details page. Uploading a file will save the fields.
System error on Submission type page
Completely fill out the contract details page.
Save as draft.
Click back into the submission
On the submission type page, clicking Continue or Save as draft will cause a system error with a 400 response from the API
Unlocking a submission, and then viewing it as a state user will cause a System error when clicking Continue or Save as draft.
When viewing a draft or unlocked submission, the submission type page will show unfilled fields for a second before filling them in with the data from the API.
These issues are probably whats causing the Cypress failures. Especially this one since we are waiting for a response. When viewing a draft or unlocked submission, the submission type page will show unfilled fields for a second before filling them in with the data from the API.. A loading animation might fix some things.

**Comment by pearl-truss** [comment]
> I manually ran through the review app and here are some things I found
> 
> * Save as draft does not save all the fields on the contract details page. Uploading a file will save the fields.
> * System error on Submission type page
>   
>   * Completely fill out the contract details page.
>   * Save as draft.
>   * Click back into the submission
>   * On the submission type page, clicking Continue or Save as draft will cause a system error with a 400 response from the API
> * Unlocking a submission, and then viewing it as a state user will cause a System error when clicking Continue or Save as draft.
> * When viewing a draft or unlocked submission, the submission type page will show unfilled fields for a second before filling them in with the data from the API.
> 
> These issues are probably what's causing the Cypress failures. Especially the last one since we are waiting for a response. A loading animation might fix some things.

I'm able to reproduce this in the review app but not locally

**Comment by JasonLin0991** [comment]
I'm getting a system error with a different console error now.
Screenshot 2024-08-15 at 3 52 47 PM

Steps to reproduce
- Start a new submission
- On the Contract details page, upload a file and click `Save as draft`
- Go back to the submission and on the Submission type page, click `Continue`
- A system error occurs
This also occurs on an unlocked submission. Is this because the rest of the form not using contract API yet and the HPP api not setting a `dateAdded`?

There is another issue where saving as draft on the Contract Details page without uploading a file does not save all the completed fields

Steps
- Start a new submission
- On the Contract details page fill out everything except for uploading a file
- Click on `Save as draft`
- Go back to the submission and on the Submission type page click `Continue`
- Only some of the fields are filled out.

**Pull request closed without merging**